src/main/kotlin/com/team4099/robot2020/subsystems/Intake.kt

```diff
@@ -49,13 +56,28 @@ object Intake : Subsystem {
     }
 
     @Synchronized
-    override fun onLoop(timestamp: Double, dt: Double) {
-        synchronized(this) {
-            when (intakeState) {
-                IntakeState.IN -> intakePower = -1.0
-                IntakeState.OUT -> intakePower = 1.0
-                IntakeState.IDLE -> intakePower = 0.0
+    override fun onLoop(timestamp: Double, dT: Double) {
+
+        if (inBeamBroken) {
+            if (inBeamBrokenTimestamp == -1.0) {
+                inBeamBrokenTimestamp = timestamp
+            }
+        } else {
+            ballCount += ((timestamp - inBeamBrokenTimestamp) / Constants.BeamBreak.IN_BEAM_BROKEN_BALL_TIME).toInt()
+            inBeamBrokenTimestamp = -1.0
+        }
+
+        @SuppressWarnings("MagicNumber")
```

Comment:
instead of doing this, define max ball count as a constant somewhere (kinda stupid but better than using the suppression I think)